anvio/drivers/hmmer.py

```diff
@@ -119,17 +132,28 @@ def run_hmmscan(self, source, alphabet, context, kind, domain, num_genes_in_mode
                             "Anvi'o will use %s process with %s cores each instead. I hope thats okay for you. " %
                              (str(self.num_threads_to_use), str(num_parts), target, str(num_parts), cores_per_process))
 
+        if alphabet in ['DNA', 'RNA'] and self.program_to_use == 'hmmsearch':
+            self.run.warning("You requested to use the program `%s`, but because you are working with %s sequences Anvi'o will use `nhmmscan` instead. "
+                            "We hope that is alright." % (self.program_to_use, alphabet))
+
 
         for part_file in self.target_files_dict[target]:
             log_file = part_file + '_log'
             output_file = part_file + '_output'
             shitty_file = part_file + '_shitty'
 
-            cmd_line = ['nhmmscan' if alphabet in ['DNA', 'RNA'] else 'hmmscan',
-                        '-o', output_file, *noise_cutoff_terms.split(),
-                        '--cpu', cores_per_process,
-                        '--tblout', shitty_file,
-                        hmm_file_path, part_file]
+            if noise_cutoff_terms:
+                cmd_line = ['nhmmscan' if alphabet in ['DNA', 'RNA'] else self.program_to_use,
+                            '-o', output_file, *noise_cutoff_terms.split(),
+                            '--cpu', cores_per_process,
+                            '--tblout', shitty_file,
+                            hmm, part_file]
+            else: # if we didn't pass any noise cutoff terms, here we don't include them in the command line
+                cmd_line = ['nhmmscan' if alphabet in ['DNA', 'RNA'] else self.program_to_use,
```

Comment:
Yes, totally. There is a warning for this in the code above. :)
```
if alphabet in ['DNA', 'RNA'] and self.program_to_use == 'hmmsearch':
            self.run.warning("You requested to use the program `%s`, but because you are working with %s sequences Anvi'o will use `nhmmscan` instead. "
                             "We hope that is alright." % (self.program_to_use, alphabet))
```